app/services/users.js

```diff
@@ -52,3 +52,18 @@ exports.findUserByMail = mail =>
   User.findOne({ where: { mail } }).catch(error => {
     throw errors.databaseError(error);
   });
+
+exports.getUserAlbums = async (userId, tokenMail) => {
+  try {
+    logger.info(`Index UserId: ${userId}'s Albums`);
+    const user = await User.findByPk(userId);
+    const tokenUser = await this.findUserByMail(tokenMail);
+    if (!(user.id === tokenUser.id || tokenUser.type === 'admin')) {
+      throw errors.authenticationError(`UserId ${userId} doesn't have the required permission`);
+    }
```

Comment:
You have a lot of logic for getting albums, you just split this logic in middleware, controllers, etc. 
The service should only query the albums of a given params